### PR TITLE
Prepare for v0.14.0

### LINKED ISSUE #1100: v0.14.0 preparation
Prepare a v0.14.0 release to mainly include numpy 2.0

What PRs **should** be merged before that? (Maintainers feel free to edit)

- #1026 
- #1039 ? This could be helpful for the release itself

These are small PRs with bug fixes, which **might** be good to include:
- #1077 
- #1065 
- #1088 
- #1069

### PR DESCRIPTION
**Pull request by BenediktBurger**
Closes #1100

@bilderbuchi I prepared the changelog for v0.14.0

The date is still missing.

### CODE REVIEW

**Comment by bilderbuchi** [comment]
LGTM. Maybe you want to call out the updated citation information/DOI in the main summary, to make academic users aware? Either way is fine by me.

**Comment by bilderbuchi** [comment]
@BenediktBurger should we do this release manually instead of waiting for the automatic workflow to be possible? #

IIUC, we should soon push another follow-up release (with updated dependency info) out after numpy 2.0 has been released, anyway (where we can test the automatic workflow)? Right, @CasperSchippers ?

**Comment by BenediktBurger** [comment]
Ok, if you can spare the time for a manual release, it's fine with me.

Note, that a few more PRs have been merged in the mean time.

**Comment by BenediktBurger** [comment]
I'd like to have #1110 merged as well (it's just reducing the time limit in a test to make it pass).

**Comment by CasperSchippers** [comment]
> IIUC, we should soon push another follow-up release (with updated dependency info) out after numpy 2.0 has been released, anyway (where we can test the automatic workflow)? Right, @CasperSchippers ?

We should indeed, after numpy 2.0 is released, update the requirements file for the numpy 2 test. I think we don't need to do much else, so I'm not sure if (as long as all the tests keep on succeeding) that by itself warrants an additional release.

**Comment by BenediktBurger** [comment]
@bilderbuchi , are you fine with releasing?
In that case, I'll add the current date and start (and accept) a release.

The tests fail due to #1112

**Comment by bilderbuchi** [comment]
You wanted to pull in #1110, is that still the case?

I'd be quite wary of merging (and especially releasing!) with broken tests!

**Comment by BenediktBurger** [comment]
> You wanted to pull in #1110, is that still the case?

I'd love to, but I need a reviewer...

> I'd be quite wary of merging (and especially releasing!) with broken tests!

Let's diagnose the setuptools thing.

**Comment by BenediktBurger** [comment]
I bumped the requirement to 8.1.0 for setuptools_scm, but mamba loads 8.0.4.
That is probably the reason, why the error persists.

**Comment by BenediktBurger** [comment]
I deleted all the caches and restarted the checks

**Comment by BenediktBurger** [comment]
With the most recent setuptools_scm version (enforced via `pymeasure.yml` environment file), the tests pass.

**Comment by BenediktBurger** [comment]
@bilderbuchi , now we can release with the tests passing, I think (today's date is already present).

File: CHANGES.rst
Comment (CasperSchippers) [inline CHANGES.rst:4]: While I don't like advertising my own work, one thing I would like to 'advertise' in this main-items section is the addition of the default queue method combined with the file-input; the consequence of this is that it should be a lot easier to get started with the PyMeasure interface. So something along the lines of:
```
- Add default :code:`queue` method and a :code:`FileInputWidget`, allowing to more quicklly get started with the PyMeasure user interface (:code:`ManagedWindow`).
```
I'll leave it up to you if you agree with this and want to add this to the "Main items" section.

(In Dutch, we have a nice saying about this, which translates to "We from Toilet-Duck recommend Toilet-Duck", but I don't thinkt there is a good English equivalent to this saying).

**Review by CasperSchippers — APPROVED** [review]
Good work.

I've only added a single suggestion.

File: CHANGES.rst
Comment (BenediktBurger) [reply CHANGES.rst:4]: It's ok to add important parts.

**Comment by BenediktBurger** [comment]
@bilderbuchi I created a release draft. Do you want to address #1112 with the `setuptools_scm` dependency first?
Feel free to release my draft and upload thus to Pypi.

Btw. I did uncheck the "latest release" on purpose, such that we can verify the pypi upload and check the "latest release" afterwards.

File: CHANGES.rst
Comment (BenediktBurger) [reply CHANGES.rst:4]: Thanks for adding your thought regarding important changes.

**Comment by bilderbuchi** [comment]
I just did!

**Comment by bilderbuchi** [comment]
Seems to have worked: https://pypi.org/project/PyMeasure/#history

Can you confirm this is all as it should be on your end?

Now we only need to merge this PR with rebase-and-merge so that the tag ends up inline on the master branch, afaik GH just does a fast-forward merge if possible.

**Comment by BenediktBurger** [comment]
Yes, I noticed it, too and was thinking about solutions.

Would a normal merge have been better, as that would add the commit (with the tag) as is to the main tree, wouldn't it?
It would be the second to last commit after the merge commit.

**Comment by BenediktBurger** [comment]
Solution for now: Merge the v0.14.0_release branch into the master?

**Comment by bilderbuchi** [comment]
> Would a normal merge have been better, as that would add the commit (with the tag) as is to the main tree, wouldn't it?
> It would be the second to last commit after the merge commit.

Yes, that's probably the least painful option. We should document that we need to choose that merge method in the release process.

**Comment by bilderbuchi** [comment]
> Solution for now: Merge the v0.14.0_release branch into the master?

yeah, although I suspect this could just be empty. IDK what will happen, try locally first, please.

**Comment by BenediktBurger** [comment]
> > Solution for now: Merge the v0.14.0_release branch into the master?
> 
> yeah, although I suspect this could just be empty/a no-op. IDK what will happen, try locally first, please.

A local try was my plan: It succeeded. Shall I push it?

**Comment by BenediktBurger** [comment]
Now the tag is shown to be part of the master branch!

**Comment by BenediktBurger** [comment]
Thanks for your collaboration on this additional issue.

**Comment by bilderbuchi** [comment]
Sure, no problem!

**Pull request merged**
